src/components/WordSelector/WordSelector.js

```diff
@@ -38,49 +47,83 @@ function WordSelector(props) {
   }
 
-  function handleClose() {
-    props.setSelectMode(false);
-  }
-
   function handleSelect() {
     props.selectWords(checkedWords);
     props.setSelectMode(false);
     props.selectGroup(props.name);
+    props.changeColor(props.index);
   }
 
+  function handleChooseAll() {
+    if (props.name !== wordGroup) {
+      updateWordGroup(props.name);
+      setDisableSelect(false);
+    }
+    if (!chooseAll) {
+      updateWords([...props.group]);
+      updateWordCount(props.group.length);
+    } else {
+      updateWords([]);
+      updateWordCount(0);
+    }
+    setChooseAll(!chooseAll);
+  }
   function wordSelection(word, index) {
     return (
-      <div className="Words" key={index}>
-        <label class="container">
-          <input
-            class="check"
-            checked={checkedWords.includes(word) ? true : null}
-            type="checkbox"
-            name={word}
-            onChange={() => handleCheck(word)}
-          />
-          <div className="word">{word}</div>
-        </label>
-      </div>
+      <Col sm="4">
+        <div className="words" key={index}>
+          <label class="container">
+            <input
+              class="check"
+              checked={checkedWords.includes(word)}
+              type="checkbox"
+              name={word}
+              onChange={() => handleCheck(word)}
+            />
+            <div className="word">{word}</div>
+          </label>
+        </div>
+      </Col>
     );
   }
 
   return (
     <div className="WordSelector">
-      <div className="GroupTitle">{props.name}</div>
-      <hr className="GroupTitleUnderline"></hr>
-      <div className="WordDisplay">{props.group.map(wordSelection)}</div>
-      <div>
-        <Button onClick={() => handleClose()} className="CloseSelection">
-          Close
-        </Button>
-        <Button
-          disabled={disableNext()}
-          onClick={() => handleSelect()}
-          className="NextButton"
-        >
-          Select
-        </Button>
+      <div ref={ref}>
+        <Row>
+          <Col sm="8">
+            <div className="GroupTitle">{props.name}</div>
+          </Col>
+          <Col sm="4">
+            <div className="SelectCheckbox">
+              <label class="container">
+                <input
+                  class="check"
+                  checked={
+                    checkedWords.length == props.group.length && !disableSelect
+                  }
+                  type="checkbox"
+                  name="Select All"
+                  onChange={handleChooseAll}
+                />
+                <div className="word">Choose All</div>
+              </label>
+            </div>
+          </Col>
+        </Row>
+        <hr className="GroupTitleUnderline"></hr>
+        <div className="WordDisplay">
+          {props.group.map((word, index) => wordSelection(word, index))}
+        </div>
+        <div className="Padding">
+          <Button
+            disabled={disableNext()}
```

Comment:
is this supposed to be `{disableNext}` or `{disableNext()}` (w the quotes) bc i know when you have the quotes it runs over and over again